### PR TITLE
[flutter_tools] support enable-experiment in flutter analyze

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Allow providing the --enable-experiment flag to the dart analyzer. This supports the eventual null-safety preview.

https://github.com/flutter/flutter/issues/52819

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart [hunk 1]
```diff
@@ -303,6 +303,32 @@ StringBuffer bar = StringBuffer('baz');
     }
   });
 
+  testUsingContext('analyze once supports analyzing null-safe code', () async {
+    const String contents = '''
+int? bar;
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart:308]: This is parse failure without the flag

File: packages/flutter_tools/test/src/fake_process_manager.dart [hunk 2]
```diff
@@ -198,7 +203,7 @@ abstract class FakeProcessManager implements ProcessManager {
       fakeCommand.onRun,
       _pid,
       fakeCommand.stderr,
-      null, // stdin
+      fakeCommand.stdin, // stdin
```
Comment (jmagman) [inline packages/flutter_tools/test/src/fake_process_manager.dart:206]: Remove comment.

**Review by jmagman — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/test/src/fake_process_manager.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/src/fake_process_manager.dart]: Done

**Pull request merged**